pkg/controllers/et/verticadbref_reconciler.go

```diff
@@ -0,0 +1,131 @@
+limitations under the License.
+*/
+
+package et
+
+import (
+	"context"
+	"fmt"
+
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	"github.com/vertica/vertica-kubernetes/pkg/etstatus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type VerticaDBRefReconciler struct {
+	VRec *EventTriggerReconciler
+	Et   *vapi.EventTrigger
+}
+
+func MakeVerticaDBRefReconciler(r *EventTriggerReconciler, et *vapi.EventTrigger) controllers.ReconcileActor {
+	return &VerticaDBRefReconciler{VRec: r, Et: et}
+}
+
+func (r *VerticaDBRefReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctrl.Result, error) {
+	for refIdx, ref := range r.Et.Spec.References {
+		if ref.Object.Kind != vapi.VerticaDBKind || ref.Object.APIVersion != vapi.GroupVersion.String() {
+			continue
+		}
+
+		// Create the refStatus object as we need to update the status at the
+		// end regardless of what happens.
+		refStatus := vapi.ETRefObjectStatus{
+			APIVersion: ref.Object.APIVersion,
+			Namespace:  ref.Object.Namespace,
+			Name:       ref.Object.Name,
+			Kind:       ref.Object.Kind,
+		}
+
+		vdb := &vapi.VerticaDB{}
+		nm := types.NamespacedName{Namespace: r.Et.Namespace, Name: ref.Object.Name}
+
+		if err := r.VRec.Client.Get(ctx, nm, vdb); err != nil {
+			if errors.IsNotFound(err) {
+				if errs := etstatus.Apply(ctx, r.VRec.Client, r.Et, &refStatus); errs != nil {
+					return ctrl.Result{}, errs
+				}
+
+				continue
+			}
+
+			return ctrl.Result{}, err
+		}
+
+		refStatus.ResourceVersion = vdb.ResourceVersion
+		refStatus.UID = vdb.GetUID()
+
+		shouldCreateJob := true
+		for _, match := range r.Et.Spec.Matches {
+			if !r.matchStatus(vdb, ref, match) {
+				shouldCreateJob = false
+				break
+			}
+		}
+
+		// Check if job already created.
+		if len(r.Et.Status.References) > 0 && r.Et.Status.References[refIdx].JobName != "" {
+			shouldCreateJob = false
+		}
+
+		if shouldCreateJob {
+			// Kick off the job
+			job, err := r.VRec.createJob(ctx, r.Et)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			refStatus.JobNamespace = job.Namespace
+			refStatus.JobName = job.Name
+		}
+
+		if err := etstatus.Apply(ctx, r.VRec.Client, r.Et, &refStatus); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// matchStatus will check if the matching condition given from the manifest
+// matches with the reference object and return false when it doesn't match.
+func (r *VerticaDBRefReconciler) matchStatus(vdb *vapi.VerticaDB, ref vapi.ETReference, match vapi.ETMatch) bool {
+	// Grab the condition based on what was given.
+	conditionType := vapi.VerticaDBConditionType(match.Condition.Type)
+	conditionTypeIndex, ok := vapi.VerticaDBConditionIndexMap[conditionType]
+	if !ok {
+		r.VRec.Log.Info(fmt.Sprintf("vertica DB condition %s missing from VerticaDBConditionType", match.Condition.Type))
```

Comment:
minor: we don't need to use sprintf here. Info() takes multiple arguments that allow you to pass in key/value pairs.
```suggestion
		r.VRec.Log.Info("vertica DB condition missing from VerticaDBConditionType", "condition", match.Condition.Type)
```